pkg/process/config/yaml_config.go

```diff
@@ -270,3 +274,22 @@ func (a *AgentConfig) setCheckInterval(ns, check, checkKey string) {
 		a.CheckIntervals[checkKey] = time.Duration(interval) * time.Second
 	}
 }
+
+// Separate handler for initializing the process discovery check.
+// Since it has its own unique object, we need to handle loading in the check config differently separately
+// from the other checks.
+func (a *AgentConfig) initProcessDiscoveryCheck() {
+	root := key(ns, "process_discovery")
+
+	// We don't need to check if the key exists since we already bound it to a default in InitConfig
+	a.CheckIntervals[DiscoveryCheckName] = config.Datadog.GetDuration(key(root, "interval"))
```

Comment:
we should probably enforce the min interval here, maybe `10m`?

Reply:
Fixed! https://github.com/DataDog/datadog-agent/pull/9082/commits/7aecc9bd43d51851990944d64170bcebc44fe8a0